js/app.jsx

```diff
@@ -166,6 +193,98 @@ var SearchableRemoteListMixin = {
 
 }
 
+var FunctionRow = React.createClass({
+  render: function() {
+    var newButton = <div></div>;
+    var diffButton = <div></div>;
+
+    var diffURL = "";
+    var diffDisabled = true;
+
+    //create diff button
+    if (this.props.setting.pagename =="caseversion") {
+      if (typeof this.props.checked !== "undefined"){
+        diffURL = "diff.html?lhs=" + this.props.checked[0] + "&rhs=" + this.props.checked[1]
+        if (this.props.checked.length == 2){
+          var diffDisabled = false;
+        }
+      }
+      diffButton = <Button bsStyle="success" target="blank_" href={diffURL} disabled={diffDisabled}>diff</Button>
+    }
+
+    //create new button
+    if (this.props.setting.pagename == "caseversion") {
+      newButton = <Button href='https://moztrap.mozilla.org/manage/case/add/' >+ New Case</Button>
+    } else if (this.props.setting.pagename == "suite") {
+      newButton = <Button bsStyle="success" href='https://moztrap.mozilla.org/manage/suite/add/' >+ New Suite</Button>
+    }
+    return (
+      <Row>
+        <Col md="12">
+          {newButton}
+          {diffButton}
+        </Col>
+      </Row>
+    )
+  }
+})
+
+
+var SearchableList = React.createClass({
+  mixins: [SearchableRemoteListMixin],
+
+  buildURL: function(query) {
+      var api_url =  config.baseUrl + "/api/v1/" + this.props.setting.pagename;
+      return (buildQueryUrl(api_url, query, caseversionCodegen) +
+              "&limit=" + config.defaultListLimit
+             );
+  },
+
+  handleQueueUpdate: function(e) {
+    var newState = {};
+    if (e.target.checked){
+      newState['checked'] = this.state['checked'].concat(e.target.value);
+    }
+    else {
+      this.state['checked'].splice(this.state['checked'].indexOf(e.target.value), 1);
+      newState['checked'] = this.state['checked'];
+    }
+    this.setState(newState);
+  },
+
+  render: function() {
+    return (
+      <div>
+        <FunctionRow setting={this.props.setting} checked={this.state.checked} />
+        <SearchForm ref="searchform" query={this.state.query} onSubmit={this.handleSearch} setting={this.props.setting}/>
+        <SearchTable setting={this.props.setting} data={this.state.data} handleAddFilter={this.handleAddFilter} handleCheck={this.handleQueueUpdate}/>
+        <MoreLink onLoadMore={this.handleLoadMore} buttonDisabled={this.state.hasNoLinkToShow}/>
+      </div>
+    )
+  }
+})
+
+// http://stackoverflow.com/questions/27864720/react-router-pass-props-to-handler-component
+var CaseVerWrapper = React.createClass({
+  render: function() {
+    return (
+      <Grid>
+      <SearchableList setting={caseverSetting}/>
```

Comment:
Can we create settings like this:
```
var settings = {
  'caseversion': {'syntax':..., 'header':...},
  'suite': {'syntax':..., 'header':...}
}
```

And we use `<SearchableList pagename={"caseversion"}/>`, in `SearchableList`, it can access the setting by `settings[this.props.pagename]`. Thus removing this wrapper.